src/grimp/adaptors/importscanner.py

```diff
@@ -14,151 +15,28 @@ class NotAnImport(Exception):
     pass
 
 
-class ImportScanner(AbstractImportScanner):
```

Comment:
What's the reason for moving this down to the bottom of the module?

As `ImportScanner` is the public facing part of the module, I'd like it to stay up the top. I tried moving it to the top and everything seemed to work fine locally, but if you run into issues we could consider adding `from __future__ import annotations` which might help.

Separate reason: it's difficult to review this PR in its current form so I'd like to be able to understand what has changed better.

Reply:
The only reason behind moving it was to avoid the furture import. I'll move it back if it's okay

Reply:
Fixed, the diff should be better now